### PR TITLE
Enabling popularProducts header hover

### PR DESCRIPTION
**Pull request by ganeshparsads**
@anantheshadiga @divyadaglia-unbxd please review.

### CODE REVIEW

File: unbxdAutosuggest.js [hunk 1]
```diff
@@ -411,6 +411,11 @@ var unbxdAutoSuggestFunction = function ($, Handlebars, undefined) {
 						return;
 					if (dataValue) {
 						var query = dataValue + (dataFiltername != '' ? ':' + dataFiltername + ':' + dataFiltervalue : '')
+						// updating product header while hovering on suggestions
+						if (self.options.filtered && self.options.popularProducts.header) {
+							self.options.popularProducts.header = self.options.popularProducts.header +
+								'&nbsp;in&nbsp;<span>' + dataValue + '</span>';
```
Comment (anantheshadiga) [inline unbxdAutosuggest.js]: Not generalised, the template of header is hard coded(Taking about `in` aspect).. How can we make more generic?

Comment (ganeshparsads) [reply unbxdAutosuggest.js]: @anantheshadiga solved this in `4c4fe11`. 
Now providing option for handleBar template in popularProduct header.

**Comment by ganeshparsads** [comment]
@anantheshadiga `compiledPopularProductHeader` is an internal field. For the SDK users option will be `popularProducts.header`.

**Comment by anantheshadiga** [comment]
But once you have added in `options`, Someone can override the parameter, isn't it?

File: tests/index.html [hunk 2]
```diff
@@ -71,7 +71,7 @@
             image: true,
             imageUrlOrFunction: 'imageUrl',
             currency: '$',
-            header: '{{{checkInputQuery}}}',
+            header: 'Popular Products in {{hoverSuggestion}}',
```
Comment (anantheshadiga) [inline tests/index.html]: Can you add test cases for this
1. WIth simple string
2. with handlebar template
3. Without any value(empty string)

**Comment by ganeshparsads** [comment]
> But once you have added in `options`, Someone can override the parameter, isn't it?

@anantheshadiga this is an internal option, not exposed to outside environment.

**Pull request merged**